File: 4186-count-residue-prefixes/count-residue-prefixes.cpp
class Solution {
public:
    int residuePrefixes(string nums) {
     unordered_set<char>s;
     int count=0;
     for(int i=0;i<nums.size();i++){
        s.insert(nums[i]);
        if((i+1)%3== s.size()) count++;
     }   
    return count;
    }
};

File: 4186-count-residue-prefixes/count-residue-prefixes.js
/**
 * @param {string} s
 * @return {number}
 */
var residuePrefixes = function(nums) {
     const s=new Set();
     var count=0;
     for(var i=0;i<nums.length;i++){
        s.add(nums[i]);
        if((i+1)%3== s.size) count++;
     }   
    return count;
};